### ISSUE TITLE
MeshRefinement runs in libmesh_assert(elem->parent())

### ISSUE THREAD

**Issue by BalticPinguin** [libMesh/libmesh#2699]
I am trying to make mesh-refinement working for me (i.e. with infinite elements).
My idea was to adapt the ``KellyErrorEstimator`` to just skip infinite elements, giving them an error of ``0`` and thus ignoring those for refinement (I am not interested in coarsening).
However, I than apply ``mesh_refinement.flag_elements_by_error_fraction()`` with setting ``mesh_refinement.coarsen_fraction()=0`` I expectated to do what I want.
But in the function, the actual check is __``else if (elem_error <= coarsen_cutoff)``__ (src/mesh/mesh_refinement_flagging.C , line150), so my elements are marked actually for coarsening.

This behaviour turns to the above-mentioned assert when running ``MeshRefinement::refine_elements``, because
 `` MeshRefinement::_smooth_flags()`` checks the coarsening-flags only when ``coarsening`` is ``true``. Thus, the coarsening-flags remain until the function ``eliminate_unrefined_patches ()`` comes to that element and tries to access its parent.

Long story short: the obvious workaround is to run  ``MeshRefinement::refine_and_coarsen_elements``, but my expectation would be to exclude equality, i.e. to use  __``else if (elem_error < coarsen_cutoff)``__ instead.
Any thoughts about this? Did I just construct that single case where the current behaviour is bad?

Many thinks in advance

**Comment by BalticPinguin** [comment]
Closing now; I think, using `MeshRefinement::refine_elements` unless there is something to coarsen is just right...
